types/utils.go

```diff
@@ -40,39 +38,53 @@ func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey)
 	return valset, privKey
 }
 
-// GetValidatorSet returns a validator set with a single validator
-// with the given key
-func GetValidatorSet(privKey ed25519.PrivKey) *cmtypes.ValidatorSet {
-	pubKey := privKey.PubKey()
-	valset := cmtypes.NewValidatorSet(
-		[]*cmtypes.Validator{
-			{PubKey: pubKey, Address: pubKey.Address(), VotingPower: 1},
-		},
-	)
-	return valset
+// BlockContext carries all necessary state for block generation, including configurations and outputs.
+type BlockContext struct {
+	Height       uint64
+	NTxs         int
+	PrivKey      ed25519.PrivKey // Input and Output option
+	ProposerAddr []byte          // Input option
+	Block        *Block          // Output: the generated block.
+	// Add other fields as necessary for inputs and outputs.
 }
 
-// GetRandomBlock returns a block with random data
+// GetRandomBlock generates a block with specified height and number of transactions.
+// It's tailored for simplicity, primarily used in test setups where additional outputs are not needed.
 func GetRandomBlock(height uint64, nTxs int) *Block {
-	block, _ := GetRandomBlockWithKey(height, nTxs, nil)
-	return block
+	ctx := BlockContext{
```

Comment:
`ctx` is almost exclusively used to reference `context.Context` types throughout the codebase, which is why my recommendation for changing the type name so that this could be `opts` or `config` which is more common.